### PR TITLE
Add Firebase support in the software

### PR DESCRIPTION
**Pull request by vishalmry**
Issue #110

### CODE REVIEW

File: goa/Classes/menu/MenuContext.cpp [hunk 1]
```diff
@@ -491,13 +505,15 @@ cocos2d::Node* MenuContext::createAvatarMenuItem(std::string const& normalImage,
             // document is ok
             
             std::vector<std::string> configs = this->split(cachedCharacterInformation, '_');
+            CCLOG("configs size %d", configs.size());
```
Comment (ankitk20) [inline goa/Classes/menu/MenuContext.cpp]: Remove CCLOG wherever you have added for debugging

File: goa/cocos2d/cocos/base/CCEventDispatcher.cpp [hunk 2]
```diff
@@ -322,6 +325,22 @@ void EventDispatcher::pauseEventListenersForTarget(Node* target, bool recursive/
...
 {
+        if (FirebaseHelper::isStarted)
+        { 
+            auto topEvent = firebase_instance.getTopEvent();
+            string targetName = firebase_instance.getTargetName(target->getDescription());
+            topEvent.first = firebase_instance.getTargetName(topEvent.first);
+            if (topEvent.first == targetName || (targetName == "numbers" && topEvent.first == "shapes") || (targetName == "library" && topEvent.first == "story-catalogue"))
+            { 
+                firebase_instance.setElapsedTime();
+                firebase_instance.logCustomEvent(targetName, "elapsed_time", difftime(firebase_instance.getElapsedTime(), topEvent.second));
```
Comment (ankitk20) [inline goa/cocos2d/cocos/base/CCEventDispatcher.cpp:341]: Add comments here

File: goa/proj.android-studio/goa/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -28,6 +28,7 @@
         <meta-data
             android:name="android.app.lib_name"
             android:value="MyGame" />
+        <meta-data android:name="firebase_analytics_collection_enabled" android:value="true" />
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/AndroidManifest.xml]: Auto format this

**Review by ankitk20 — CHANGES_REQUESTED** [review]

File: goa/cocos2d/cocos/base/CCEventDispatcher.cpp (hunk 2 above)
Comment (vishalmry) [reply goa/cocos2d/cocos/base/CCEventDispatcher.cpp:341]: Added the comments.

**Comment by vishalmry** [comment]
|PR checklist|Status|
|-----|:------:|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|NA|
|Is code analysis task executed and issues resolved in the CPP/native code?|No|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|No|
|Is code commenting done properly for the changes made?|Yes|
|Is correct variable naming convention followed?|Yes|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|Yes|
|Are the message strings used in the application taken from strings.xml file?|NA|
|Are the permissions which are not required removed from AndroidManifest.java file?|NA|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|NA|
|Are the release signing configurations removed from the gradle files?|NA|
|Is commented code removed from all files?|Yes|
|Are updated images compressed (< 100 KB)?|NA|
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

**Review by ankitk20 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**